<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>982</width>
    <height>818</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_5">
    <item row="0" column="0">
     <widget class="QSplitter" name="splitter_2">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <widget class="QWidget" name="widget_3" native="true">
       <layout class="QGridLayout" name="gridLayout">
        <property name="leftMargin">
         <number>9</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>9</number>
        </property>
        <property name="bottomMargin">
         <number>9</number>
        </property>
        <property name="horizontalSpacing">
         <number>6</number>
        </property>
        <item row="0" column="0">
         <widget class="QLabel" name="label">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>80</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="text">
           <string>Device Preset:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QComboBox" name="comboBox_device_preset"/>
        </item>
        <item row="1" column="0" colspan="2">
         <widget class="QLabel" name="label_6">
          <property name="maximumSize">
           <size>
            <width>16777215</width>
            <height>19</height>
           </size>
          </property>
          <property name="font">
           <font>
            <pointsize>12</pointsize>
            <weight>75</weight>
            <bold>true</bold>
           </font>
          </property>
          <property name="text">
           <string>Devices</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0" colspan="2">
         <widget class="QWidget" name="widget_9" native="true">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>300</height>
           </size>
          </property>
          <layout class="QGridLayout" name="gridLayout_12">
           <property name="leftMargin">
            <number>0</number>
           </property>
           <property name="topMargin">
            <number>0</number>
           </property>
           <property name="rightMargin">
            <number>0</number>
           </property>
           <property name="bottomMargin">
            <number>0</number>
           </property>
           <item row="0" column="0">
            <widget class="QSplitter" name="splitter">
             <property name="frameShape">
              <enum>QFrame::NoFrame</enum>
             </property>
             <property name="lineWidth">
              <number>8</number>
             </property>
             <property name="midLineWidth">
              <number>8</number>
             </property>
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="opaqueResize">
              <bool>true</bool>
             </property>
             <property name="handleWidth">
              <number>5</number>
             </property>
             <widget class="QWidget" name="widget" native="true">
              <property name="minimumSize">
               <size>
                <width>0</width>
                <height>200</height>
               </size>
              </property>
              <layout class="QGridLayout" name="gridLayout_11">
               <property name="leftMargin">
                <number>0</number>
               </property>
               <property name="topMargin">
                <number>0</number>
               </property>
               <property name="rightMargin">
                <number>0</number>
               </property>
               <property name="bottomMargin">
                <number>0</number>
               </property>
               <item row="0" column="2">
                <widget class="QPushButton" name="pushButton_remove_device">
                 <property name="maximumSize">
                  <size>
                   <width>30</width>
                   <height>16777215</height>
                  </size>
                 </property>
                 <property name="text">
                  <string>-</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="1">
                <widget class="QPushButton" name="pushButton_add_device">
                 <property name="maximumSize">
                  <size>
                   <width>30</width>
                   <height>16777215</height>
                  </size>
                 </property>
                 <property name="text">
                  <string>+</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="0">
                <widget class="QLineEdit" name="lineEdit_device_search">
                 <property name="placeholderText">
                  <string>Search</string>
                 </property>
                </widget>
               </item>
               <item row="3" column="0" rowspan="2" colspan="3">
                <widget class="QTreeView" name="treeView_devices"/>
               </item>
              </layout>
             </widget>
             <widget class="QWidget" name="widget_2" native="true">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimumSize">
               <size>
                <width>0</width>
                <height>0</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>16777215</width>
                <height>16777215</height>
               </size>
              </property>
              <layout class="QGridLayout" name="gridLayout_2">
               <property name="leftMargin">
                <number>0</number>
               </property>
               <property name="topMargin">
                <number>0</number>
               </property>
               <property name="rightMargin">
                <number>0</number>
               </property>
               <property name="bottomMargin">
                <number>0</number>
               </property>
               <item row="0" column="0">
                <widget class="QPushButton" name="pushButton_make_EPICS_environment">
                 <property name="text">
                  <string>Make EPICS IOC</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="1">
                <widget class="QPushButton" name="pushButton_show_console_output">
                 <property name="text">
                  <string>Show console output</string>
                 </property>
                </widget>
               </item>
               <item row="2" column="0" colspan="2">
                <widget class="QTextEdit" name="textEdit_console_output"/>
               </item>
               <item row="1" column="0" colspan="2">
                <widget class="QProgressBar" name="progressBar_devices">
                 <property name="enabled">
                  <bool>true</bool>
                 </property>
                 <property name="value">
                  <number>100</number>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
             <widget class="QWidget" name="device_config_widget" native="true">
              <layout class="QGridLayout" name="gridLayout_7">
               <property name="leftMargin">
                <number>0</number>
               </property>
               <property name="topMargin">
                <number>0</number>
               </property>
               <property name="rightMargin">
                <number>0</number>
               </property>
               <property name="bottomMargin">
                <number>0</number>
               </property>
              </layout>
             </widget>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="widget_4" native="true">
       <layout class="QGridLayout" name="gridLayout_4">
        <item row="0" column="0">
         <widget class="QLabel" name="label_2">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>110</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="text">
           <string>Measurement Preset:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QComboBox" name="comboBox_measurement_preset"/>
        </item>
        <item row="1" column="0" colspan="2">
         <widget class="QWidget" name="widget_5" native="true">
          <layout class="QGridLayout" name="gridLayout_3">
           <property name="leftMargin">
            <number>0</number>
           </property>
           <property name="topMargin">
            <number>0</number>
           </property>
           <property name="rightMargin">
            <number>0</number>
           </property>
           <property name="bottomMargin">
            <number>0</number>
           </property>
           <item row="0" column="0">
            <widget class="QSplitter" name="splitter_3">
             <property name="frameShape">
              <enum>QFrame::NoFrame</enum>
             </property>
             <property name="lineWidth">
              <number>10</number>
             </property>
             <property name="midLineWidth">
              <number>10</number>
             </property>
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <widget class="QWidget" name="widget_6" native="true">
              <layout class="QGridLayout" name="gridLayout_6">
               <property name="leftMargin">
                <number>9</number>
               </property>
               <property name="topMargin">
                <number>0</number>
               </property>
               <property name="rightMargin">
                <number>9</number>
               </property>
               <property name="bottomMargin">
                <number>0</number>
               </property>
               <item row="2" column="0">
                <widget class="QWidget" name="widget_10" native="true">
                 <layout class="QGridLayout" name="gridLayout_15">
                  <property name="leftMargin">
                   <number>0</number>
                  </property>
                  <property name="topMargin">
                   <number>0</number>
                  </property>
                  <property name="rightMargin">
                   <number>0</number>
                  </property>
                  <property name="bottomMargin">
                   <number>0</number>
                  </property>
                  <property name="spacing">
                   <number>0</number>
                  </property>
                  <item row="0" column="0">
                   <widget class="QSplitter" name="splitter_4">
                    <property name="orientation">
                     <enum>Qt::Vertical</enum>
                    </property>
                    <widget class="QWidget" name="widget_11" native="true">
                     <layout class="QGridLayout" name="gridLayout_13">
                      <property name="leftMargin">
                       <number>0</number>
                      </property>
                      <property name="topMargin">
                       <number>0</number>
                      </property>
                      <property name="rightMargin">
                       <number>0</number>
                      </property>
                      <property name="bottomMargin">
                       <number>0</number>
                      </property>
                      <item row="0" column="1">
                       <widget class="QPushButton" name="pushButton_add_protocol">
                        <property name="maximumSize">
                         <size>
                          <width>30</width>
                          <height>16777215</height>
                         </size>
                        </property>
                        <property name="text">
                         <string>+</string>
                        </property>
                       </widget>
                      </item>
                      <item row="0" column="2">
                       <widget class="QPushButton" name="pushButton_remove_protocol">
                        <property name="maximumSize">
                         <size>
                          <width>30</width>
                          <height>16777215</height>
                         </size>
                        </property>
                        <property name="text">
                         <string>-</string>
                        </property>
                       </widget>
                      </item>
                      <item row="0" column="0">
                       <widget class="QLineEdit" name="lineEdit_protocol_search">
                        <property name="placeholderText">
                         <string>Search</string>
                        </property>
                       </widget>
                      </item>
                      <item row="1" column="0" colspan="3">
                       <widget class="QListView" name="listView_protocols"/>
                      </item>
                     </layout>
                    </widget>
                    <widget class="QWidget" name="widget_12" native="true">
                     <layout class="QGridLayout" name="gridLayout_14">
                      <property name="leftMargin">
                       <number>0</number>
                      </property>
                      <property name="topMargin">
                       <number>0</number>
                      </property>
                      <property name="rightMargin">
                       <number>0</number>
                      </property>
                      <property name="bottomMargin">
                       <number>0</number>
                      </property>
                      <item row="1" column="0" colspan="2">
                       <widget class="QProgressBar" name="progressBar_protocols">
                        <property name="value">
                         <number>100</number>
                        </property>
                       </widget>
                      </item>
                      <item row="0" column="0">
                       <widget class="QPushButton" name="pushButton_build_protocol">
                        <property name="text">
                         <string>Build Protocol</string>
                        </property>
                       </widget>
                      </item>
                      <item row="0" column="1">
                       <widget class="QPushButton" name="pushButton_show_output_meas">
                        <property name="text">
                         <string>Show output</string>
                        </property>
                       </widget>
                      </item>
                      <item row="2" column="0" colspan="2">
                       <widget class="QTextEdit" name="textEdit_console_output_meas"/>
                      </item>
                     </layout>
                    </widget>
                   </widget>
                  </item>
                 </layout>
                </widget>
               </item>
               <item row="0" column="0">
                <widget class="QLabel" name="label_3">
                 <property name="font">
                  <font>
                   <pointsize>12</pointsize>
                   <weight>75</weight>
                   <bold>true</bold>
                  </font>
                 </property>
                 <property name="text">
                  <string>Protocols</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="0">
                <widget class="QPushButton" name="pushButton_run_protocol">
                 <property name="text">
                  <string>Run selected protocol(s)</string>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
             <widget class="QWidget" name="widget_7" native="true">
              <layout class="QGridLayout" name="gridLayout_8">
               <property name="topMargin">
                <number>0</number>
               </property>
               <property name="bottomMargin">
                <number>0</number>
               </property>
               <item row="4" column="1">
                <widget class="QPushButton" name="pushButton_move_step_out">
                 <property name="text">
                  <string>move out</string>
                 </property>
                </widget>
               </item>
               <item row="3" column="1">
                <widget class="QPushButton" name="pushButton_move_step_in">
                 <property name="text">
                  <string>move in</string>
                 </property>
                </widget>
               </item>
               <item row="3" column="0">
                <widget class="QPushButton" name="pushButton_move_step_up">
                 <property name="text">
                  <string>move up</string>
                 </property>
                </widget>
               </item>
               <item row="4" column="0">
                <widget class="QPushButton" name="pushButton_move_step_down">
                 <property name="text">
                  <string>move down</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="0">
                <widget class="QLabel" name="label_4">
                 <property name="maximumSize">
                  <size>
                   <width>16777215</width>
                   <height>19</height>
                  </size>
                 </property>
                 <property name="font">
                  <font>
                   <pointsize>12</pointsize>
                   <weight>75</weight>
                   <bold>true</bold>
                  </font>
                 </property>
                 <property name="text">
                  <string>Sequence</string>
                 </property>
                </widget>
               </item>
               <item row="6" column="0" colspan="3">
                <widget class="QTreeView" name="treeView_protocol_sequence"/>
               </item>
               <item row="1" column="0" colspan="3">
                <widget class="QPushButton" name="pushButton_show_protocol_settings">
                 <property name="text">
                  <string>Show General Settings</string>
                 </property>
                </widget>
               </item>
               <item row="3" column="2">
                <widget class="QToolButton" name="toolButton_add_step">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="minimumSize">
                  <size>
                   <width>0</width>
                   <height>23</height>
                  </size>
                 </property>
                 <property name="text">
                  <string>+</string>
                 </property>
                </widget>
               </item>
               <item row="4" column="2">
                <widget class="QPushButton" name="pushButton_remove_step">
                 <property name="text">
                  <string>-</string>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
             <widget class="QWidget" name="widget_8" native="true">
              <layout class="QGridLayout" name="gridLayout_9">
               <property name="topMargin">
                <number>0</number>
               </property>
               <property name="rightMargin">
                <number>9</number>
               </property>
               <property name="bottomMargin">
                <number>0</number>
               </property>
               <item row="0" column="0">
                <widget class="QLabel" name="label_5">
                 <property name="maximumSize">
                  <size>
                   <width>16777215</width>
                   <height>19</height>
                  </size>
                 </property>
                 <property name="font">
                  <font>
                   <pointsize>12</pointsize>
                   <weight>75</weight>
                   <bold>true</bold>
                  </font>
                 </property>
                 <property name="text">
                  <string>Configuration</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="0">
                <widget class="QWidget" name="loopstep_configuration_widget" native="true">
                 <layout class="QGridLayout" name="gridLayout_10">
                  <property name="leftMargin">
                   <number>0</number>
                  </property>
                  <property name="topMargin">
                   <number>0</number>
                  </property>
                  <property name="rightMargin">
                   <number>0</number>
                  </property>
                  <property name="bottomMargin">
                   <number>0</number>
                  </property>
                 </layout>
                </widget>
               </item>
              </layout>
             </widget>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>982</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionSave_Device_Preset"/>
    <addaction name="actionSave_Measurement_Preset"/>
    <addaction name="actionOpen_Backup_Device_Preset"/>
    <addaction name="actionLoad_Backup_Measurement_Preset"/>
   </widget>
   <widget class="QMenu" name="menuPreferences">
    <property name="title">
     <string>Settings</string>
    </property>
    <addaction name="actionAutosave_on_closing"/>
    <addaction name="actionDark_Mode"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
   </widget>
   <widget class="QMenu" name="menuTools">
    <property name="title">
     <string>Tools</string>
    </property>
    <addaction name="actionDevice_Driver_Builder"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuPreferences"/>
   <addaction name="menuTools"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionPresets">
   <property name="text">
    <string>Device-Presets</string>
   </property>
  </action>
  <action name="actionOptions">
   <property name="text">
    <string>Options</string>
   </property>
  </action>
  <action name="actionSave_Device_Preset">
   <property name="text">
    <string>Save Device Preset</string>
   </property>
  </action>
  <action name="actionLoad">
   <property name="text">
    <string>Open</string>
   </property>
  </action>
  <action name="actionMeasurement_Presets">
   <property name="text">
    <string>Measurement-Presets</string>
   </property>
  </action>
  <action name="actionSave_Measurement_Preset">
   <property name="text">
    <string>Save Measurement Preset</string>
   </property>
  </action>
  <action name="actionOpen_Backup_Device_Preset">
   <property name="text">
    <string>Load Backup Device Preset</string>
   </property>
  </action>
  <action name="actionLoad_Backup_Measurement_Preset">
   <property name="text">
    <string>Load Backup Measurement Preset</string>
   </property>
  </action>
  <action name="actionAutosave_on_closing">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Autosave on closing</string>
   </property>
  </action>
  <action name="actionDevice_Driver_Builder">
   <property name="text">
    <string>Device Driver Builder</string>
   </property>
  </action>
  <action name="actionDark_Mode">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Dark Mode</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
